### ISSUE TITLE
Have you used batch normalization in cMIL and retrain steps

### ISSUE THREAD

**Issue by RichardChangCA** [ThoroughImages/CAMEL#9]
Thanks for your excellent work, recently I am re-implementing your work but meet some problems.

Have you used batch normalization in cMIL and retrain steps, since the batch arrangement is different in cMIL and in retrain step.

Looking forward to your reply.

Thank you.

**Comment by physicso** [comment]
Hi Rinn,
We only used BN in retrain :)
Best.

**Comment by RichardChangCA** [comment]
Thanks,

But when I remove batch norm in cMIL part, over-fitting is another issue. How do you deal with it.

Thanks again.

**Comment by thuxugang** [comment]
Hi, 
Thanks for your interest. In cMIL part, batch=1 may lead to unstable result. Therefore, we use the mean value of the losses from multiple GPUs  (4-GPU) to update the model. Also, we use the standard resnet_v2_50 without further modification, so batch norm layers exist in our model.
Best,

**Comment by RichardChangCA** [comment]
Thanks for your response.

I got it now, I only have one GPU now, I have to fix this issue by other methods.

Best,

**Comment by thuxugang** [comment]
In fact, you can use batch>1 in cMIL part. But it will be little tricky in the patch selection part. Fortunately, I had 4 GPUs at the time :)

**Comment by RichardChangCA** [comment]
Yes, there is a big issue when I re-implement the code: if I cannot train cMIL part well, the generated label for the retrain part is so bad, which inducing the retain part cannot learn well due to the garbage ground-truth. Anyway, the tricky part may be tackled based on this sentence: "Noted that we discard those poten-tially confusing samples whose predicted labels are differ-ent from their corresponding image-level labels." , if I can train cMIL well, right? Thank you!

**Comment by thuxugang** [comment]
I think you may find some related information in the previous closed issues. The accuracy of cMIL is crucial. For now, I think you may need to re-implement the code so that batch>1 can be used. In our experiment, batch=1 may lead to unstable result.

**Comment by RichardChangCA** [comment]
Yes, I will do that. Thanks for your help.

Best,
